eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Serialisation/EraIndepGenerators.hs

```diff
@@ -15,59 +16,25 @@
 
 module Test.Cardano.Ledger.Shelley.Serialisation.EraIndepGenerators
   ( genCoherentBlock,
-    genHash,
-    genShelleyAddress,
-    genByronAddress,
```

Comment:
What is the best  way then to generate a different, more specific Gen , as I did in my [PR](https://github.com/input-output-hk/cardano-ledger/pull/3174/commits/e06d141c708df0222270972e3c8998a9406bd614#diff-d2955d72f315240794b22235ed710e10b4896b47c5d39af211f37a81a270e2f4R348)?  
Perhaps I can keep this specialized function alongside the arbitrary instance, or what is more idiomatic to do?

Reply:
Yes, in that case you generate a value with a specialized function and use that instead. Hopefully with short documentation describing why this is necessary.

Ideally Arbitrary instance suppose to explore all possible states a type can be in.